java/lang/StringBuilder/StringBuilderTest.cpp

```diff
@@ -0,0 +1,399 @@
+    ASSERT_EQUAL(-1, stringBuilder.lastIndexOf(String("Vietnam"), 12));
+    ASSERT_EQUAL(11, stringBuilder.lastIndexOf((const string)"Vietnam", stringBuilder.length()));
+    ASSERT_EQUAL(11, stringBuilder.lastIndexOf(String("Vietnam"), stringBuilder.length()));
+}
+
+TEST(JavaLang, StringBuilderLength) {
+    String aString("Hello! I'm a String");
+    StringBuilder stringBuilder(aString);
+    ASSERT_EQUAL(aString.length(), stringBuilder.length());
+}
+
+TEST(JavaLang, StringBuilderReplace) {
+    StringBuilder stringBuilder((const string)"ABCXYZ");
+    ASSERT_STR("ABCXYZ", stringBuilder.toString().toString());
+    stringBuilder.replace(1, 2, (const string)"123");
+    ASSERT_STR("A123CXYZ", stringBuilder.toString().toString());
+    stringBuilder.replace(0, 100, String("Clear!"));
+    ASSERT_STR("Clear!", stringBuilder.toString().toString());
+}
+
+TEST(JavaLang, StringBuilderReverse) {
+    StringBuilder stringBuilder1((const string)"ABCXYZ");
+    stringBuilder1.reverse();
+    ASSERT_STR("ZYXCBA", stringBuilder1.toString().toString());
+
+    // Reverses all valid surrogate pairs are produced in reverse process.
+    StringBuilder stringBuilder2((const string)"\u000DC00\u000D800");
+    ASSERT_STR("\u000DC00\u000D800", stringBuilder2.toString().toString());
+    stringBuilder1.reverse();
+    ASSERT_STR("\u000DC00\u000D800", stringBuilder2.toString().toString());
+}
+
+TEST(JavaLang, StringBuilderSetCharAt) {
+    StringBuilder stringBuilder((const string)"ABCXYZ");
+    ASSERT_STR("ABCXYZ", stringBuilder.toString().toString());
+    stringBuilder.setCharAt(0, 'T');
+    ASSERT_STR("TBCXYZ", stringBuilder.toString().toString());
+    stringBuilder.setCharAt(1, Character('V'));
+    ASSERT_STR("TVCXYZ", stringBuilder.toString().toString());
+}
+
+TEST(JavaLang, StringBuilderSetLength) {
+    StringBuilder stringBuilder((const string)"Hello! I'm a StringBuilder");
+    ASSERT_STR("Hello! I'm a StringBuilder", stringBuilder.toString().toString());
+    ASSERT_EQUAL(26, stringBuilder.length());
+    stringBuilder.setLength(30);
+    ASSERT_EQUAL(30, stringBuilder.length());
+    ASSERT_STR("Hello! I'm a StringBuilder", stringBuilder.toString().toString());
+    stringBuilder.setLength(6);
+    ASSERT_EQUAL(6, stringBuilder.length());
+    ASSERT_STR("Hello!", stringBuilder.toString().toString());
+}
+
+TEST(JavaLang, StringBuilderSubString) {
+    StringBuilder stringBuilder((const string)"Hello Vietnam");
+    ASSERT_STR("Vietnam", stringBuilder.substring(6).toString());
+    ASSERT_STR("llo", stringBuilder.substring(2, 5).toString());
+    try {
+        stringBuilder.substring(-999);
+    }
+    catch (Exception e) {
+        ASSERT_STR("String index out of range: -999", e.getMessage().toString());
+    }
+}
+
+TEST(JavaLang, StringBuilderToString) {
+    String aString = "Hello!";
+    StringBuilder stringBuilder(aString.toString());
+    ASSERT_TRUE(aString == stringBuilder.toString());
```

Comment:
Please use ASSERT_STR.
If this test fail, it will show the expect and actual result, easy to compare.
ASSERT_TRUE only show **should be true**